### PR TITLE
Add setup intents support to stripe dashboard link helper

### PR DESCRIPTION
**Pull request by elia**
## Summary

We added payment and setup intents classes attached to the order and we should provide dashboard links in accordance to it.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: app/models/solidus_stripe/payment_method.rb [hunk 1]
```diff
@@ -164,11 +164,19 @@ def payment_profiles_supported?
     end
 
     def stripe_dashboard_url(payment)
-      # TODO: handle when the payment doesn't exist yet in Stripe, but we only have the setup intent
-      intent_id = payment.transaction_id
       path_prefix = '/test' if preferred_test_mode
 
-      "https://dashboard.stripe.com#{path_prefix}/payments/#{intent_id}"
+      intent_id =
+        payment.transaction_id ||
```
Comment (waiting-for-dev) [inline app/models/solidus_stripe/payment_method.rb:170]: Hey, @elia, haven't we stopped storing the intent id on the payment model? :thinking:

Comment (elia) [reply app/models/solidus_stripe/payment_method.rb:170]: No, we still provide it as the transaction id in the gateway responses, for setup intents solidus will attach it to the payment. No strong feelings, but this is the most "solidus compliant" way I think.

Comment (waiting-for-dev) [reply app/models/solidus_stripe/payment_method.rb:170]: Oh, right. Thanks for clarifying.

File: app/models/solidus_stripe/payment_method.rb [hunk 2]
```diff
@@ -164,11 +164,19 @@ def payment_profiles_supported?
...
+        SolidusStripe::PaymentIntent.where(order: payment.order).pick(:stripe_payment_intent_id) ||
+        SolidusStripe::SetupIntent.where(order: payment.order).pick(:stripe_setup_intent_id)
```
Comment (waiting-for-dev) [inline app/models/solidus_stripe/payment_method.rb:172]: Thoughts on extracting the logic to the classes? Like:

```ruby
class SolidusStripe::PaymentIntent
  def self.stripe_id(order)
    where(order: order).pick(:stripe_payment_intent_id)
  end
```

That could help duck-typing at a later stage if we're not ready yet.

File: app/models/solidus_stripe/payment_method.rb [hunk 3]
```diff
@@ -164,11 +164,19 @@ def payment_profiles_supported?
...
+
+      case intent_id
+      when /^pi_/
+        "https://dashboard.stripe.com#{path_prefix}/payments/#{intent_id}"
+      when /^seti_/
+        "https://dashboard.stripe.com#{path_prefix}/setup_intents/#{intent_id}"
```
Comment (waiting-for-dev) [inline app/models/solidus_stripe/payment_method.rb:178]: Similarly, we could duck type here, although the problem is the `payment.transaction_id` getting in the way.

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @elia! I left a couple of suggestions, although feel free to go ahead if you think that's premature optimization before we have a more defined interface.

**Comment by waiting-for-dev** [comment]
@elia, linting is failing, though :slightly_smiling_face:

**Comment by elia** [comment]
> @elia, linting is failing, though 🙂

@waiting-for-dev yeah, I dared to skip the draft at my own peril 😅 – fixing it 👍

**Comment by elia** [comment]
@waiting-for-dev I'll keep the suggestions for the strategy class, for this one I just wanted to have a working link for manual tests 🙌

**Pull request merged**